### PR TITLE
Split server up into manager + executor paradigm

### PR DESCRIPTION
**Pull request by joeyballentine**
A draft PR for what we've been discussing on Discord. Not ready to actually be merged yet.

This separates the server into two parts: The manager server which handles dependency tracking and installation, and the executor server, which handles package/node loading.

This is necessary to allow us to update and uninstall dependencies on the backend, due to the to-be-modified dependencies needing to not be loaded in python. With this new paradigm, the executor server can be killed while the manager server can stay open and manage the dependencies properly, then start the executor server back up. This also would allow us to implement the "timeout killing" feature backend-side rather than frontend-side, decoupling that even more.

The manager server acts as a proxy for most requests, while other requests are handled by it.

This does not actually implement the aforementioned dependency management and executor-killing features. Those will be tackled in a separate future PR once this one is finalized. This PR is an attempt to mimic current behavior as closely as possible.

Before being considered ready, this PR needs to be heavily tested, including through brand-new installs, to ensure no regressions.

TODO:

- [x] Revert the `requests` dependency change
- [x] Heavily test
- [x] fix CI

### CODE REVIEW

File: backend/src/dependencies/install_server_deps.py [hunk 1]
```diff
@@ -35,69 +35,87 @@
...
         "from_file": "tracerite-1.1.0-py3-none-any.whl",
+        "extra_index_url": None,
     },
     # Sanic's downstream deps that we want to pin anyway
     {
         "package_name": "websockets",
         "display_name": "websockets",
         "version": "11.0.3",
         "from_file": None,
+        "extra_index_url": None,
```
[the remark this answers is not in the document]
Comment (joeyballentine) [reply backend/src/dependencies/install_server_deps.py:82]: Sure. I didn't know that was a thing

[the remark this answers is not in the document]
Comment (joeyballentine) [reply backend/src/dependencies/install_server_deps.py:82]: We can't do this as NotRequired requires typing extensions, which we cannot assume is installed yet, but is needed for the dependency install list with that change. It's a catch 22

**Comment by joeyballentine** [comment]
New issue discovered: Video iteration can fail/freeze when FFMPEG outputs to stdout. I currently have no idea how to resolve this and i would consider it a blocker for merging.

File: backend/src/api/api.py [hunk 2]
```diff
@@ -255,6 +256,17 @@ def to_dict(self):
...
 
+    @staticmethod
+    def from_dict(data: dict[str, Any]) -> Dependency:
+        return Dependency(
+            display_name=data["displayName"],
+            pypi_name=data["pypiName"],
+            version=data["version"],
+            size_estimate=data["sizeEstimate"],
+            auto_update=data["autoUpdate"],
+            extra_index_url=data.get("findLink"),
```
[the remark this answers is not in the document]
Comment (joeyballentine) [reply backend/src/api/api.py:267]: I'm pretty sure I did this because it's not actually always defined, but I'll double check that

**Review by RunDevelopment — APPROVED** [review]
Feel free to merge when you think this is ready. The general approach and architecture look good. If it works, then it's ready.

**Pull request merged**